documentation/source/coroutines.rst

```diff
@@ -202,3 +143,53 @@ type of generator function that can be iterated with ``async for``:
 
 More details on this type of generator can be found in :pep:`525`.
 
+
+.. _yield-syntax:
+
+Generator-based coroutines
+==========================
+
+.. note:: This style is no longer recommended and support may someday be removed.
+
+Prior to Python 3.5, and the introduction of :keyword:`async` and :keyword:`await`, coroutines were implemented as wrappers around generators.
+Coroutine functions would be decorated with :class:`~cocotb.decorators.coroutine` and would use :keyword:`yield` to block on other coroutines or triggers.
+
+.. code-block:: python3
+
+    @cocotb.coroutine
+    def wait_10ns():
+        cocotb.log.info("About to wait for 10 ns")
+        yield Timer(10, units='ns')
+        cocotb.log.info("Simulation time has advanced by 10 ns")
+
+    @cocotb.coroutine
+    def wait_100ns():
+        for _ in range(10):
+            yield wait_10ns()
+
+You may see existing code that uses this syntax for coroutines, but do not worry, it is compatible with async coroutines.
+Any place that will accept async coroutines, will also accept generator-based coroutines;
+including :func:`~cocotb.fork`.
+Likewise, any object that can be used in an :keyword:`await` statement can also be used in a :keyword:`yield` statement while in a generator-based coroutine;
+including triggers like :class:`~cocotb.triggers.Timer`.
+You may also see syntax like ``yield [trigger_a, tigger_b, ...]``, which is syntactic sugar for :class:`~cocotb.triggers.First`.
```

Comment:
I think this sentence should be below the example, since it seems to me the example is associated with the previous sentence.

Reply:
I also added a small example that is the example from `First`'s docstring, but using `yield []`.